packages/celer-web-app/src/ui/frames/AppFrame.tsx

```diff
@@ -69,29 +63,76 @@ export const AppFrame: React.FC<EmptyObject> = ()=>{
 						{contextMenuRef === splitSettingsMenuItemRef && <div className={styles.submenu} style={{
 							bottom: `calc( 100vh - ${contextMenuRef.current?.getBoundingClientRect().bottom || 0}px )`,
 						}}>
-							<MenuItemWithValue value={getSplitSettingText(splitSetting[SplitType.Shrine])} action={function (): void {
-								setSplitSetting(!splitSetting[SplitType.Shrine], SplitType.Shrine);
+							<MenuItemWithValue value={getSplitSettingText(setting.splitSettings[SplitType.Shrine])} action={function (): void {
+								setSetting({
+									...setting,
+									splitSettings: {
+										...setting.splitSettings,
+										[SplitType.Shrine]: !setting.splitSettings[SplitType.Shrine]
+									}
+								});
```

Comment:
Can we use `immer` for manipulating immutable objects? https://immerjs.github.io/immer/